plugins/org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/ui/quickfix/QuickfixProviderFragment2.xtend

```diff
@@ -0,0 +1,177 @@
+/*******************************************************************************
+ * Copyright (c) 2015 itemis AG (http://www.itemis.eu) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *******************************************************************************/
+package org.eclipse.xtext.xtext.generator.ui.quickfix
+
+import javax.inject.Inject
+import org.eclipse.xtend.lib.annotations.Accessors
+import org.eclipse.xtext.Grammar
+import org.eclipse.xtext.xtext.generator.AbstractGeneratorFragment2
+import org.eclipse.xtext.xtext.generator.CodeConfig
+import org.eclipse.xtext.xtext.generator.XtextGeneratorNaming
+import org.eclipse.xtext.xtext.generator.model.FileAccessFactory
+import org.eclipse.xtext.xtext.generator.model.GuiceModuleAccess
+import org.eclipse.xtext.xtext.generator.model.TypeReference
+import org.eclipse.xtext.xtext.generator.validation.ValidatorFragment2
+
+import static extension org.eclipse.xtext.xtext.generator.util.GrammarUtil2.*
+
+/**
+ * Contributes the Quickfix provider stub, either in Xtend or Java language.
+ * 
+ * @author Christian Schneider - Initial contribution and API
+ */
+class QuickfixProviderFragment2 extends AbstractGeneratorFragment2 {
+
+	@Inject
+	extension XtextGeneratorNaming
+
+	@Inject
+	extension CodeConfig
+	
+	@Inject
+	extension ValidatorFragment2
```

Comment:
This is not the same ValidatorFragment that the user configured. Maybe extract a class `ValidatorNaming` that the user can bind in the guice module?